pkg/filters/fields.go

```diff
@@ -81,15 +81,51 @@ type FieldFilter struct {
 	fields         fmutils.NestedMask
 	action         tetragon.FieldFilterAction
 	invertEventSet bool
+	needsCopy      map[tetragon.EventType]struct{}
 }
 
 // NewFieldFilter constructs a new FieldFilter from a set of fields.
 func NewFieldFilter(eventSet []tetragon.EventType, fields []string, action tetragon.FieldFilterAction, invertEventSet bool) *FieldFilter {
+	var maybeNeedsCopy bool
+	if action == tetragon.FieldFilterAction_EXCLUDE {
+		for _, field := range fields {
+			if field == "process.pid" || field == "parent.pid" {
+				maybeNeedsCopy = true
+				break
+			}
+		}
+	} else if action == tetragon.FieldFilterAction_INCLUDE {
```

Comment:
Can you explain the logic behind the INCLUDE?

I understand the EXCLUDE as follows: We need to copy if we need to remove fields from (and thus modify)  process or parent. Why do we need a copy on include? Is the idea that fields would have been excluded at the top level so unless we have both, something will be deleted?

Again, I'm not sure why we are only checking for .pid (see above). Is this the only field that can be included/excluded from .process and .parent?

Reply:
@kkourt think of include like the inverse of exclude. If the fields we care about are NOT included they will be excluded. We're checking for that here.